### PR TITLE
ICRC-29: Window Post Message Transport for ICRC-25

### PR DESCRIPTION
**Pull request by frederikrothenberger**
This PR adds ICRC-29. This is the last standard required for  the minimal signer interaction. It defines one possible implementation of an ICRC-25 compliant transport to send message from the relying party to the signer.

### CODE REVIEW

File: topics/icrc_29_window_post_message_transport.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+This standard defines a transport channel to send ICRC-25 messages from a relying party to a signer. The transport channel is based on the [window.postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) API.
+
+## Terminology
+
+* signer: A service that manages a user's keys and can sign and perform canister calls on their behalf.
+* relying party: A service that wants to request calls on a specific canister.
+
+## Trust Assumption
+For this standard to represent an [ICRC-25](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_25_signer_interaction_standard.md) compliant transport channel, the following assumptions must hold:
+* The users machine is not compromised. In particular, the browser must deliver messages unchanged and represent the `origin` of the different parties correctly.
```
Comment (przydatek) [inline topics/icrc_29_window_post_message_transport.md:18]: "user's machine"?

File: topics/icrc_29_window_post_message_transport.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+* DNS entries are not compromised. The users machine must be able to resolve the domain names of the relying party and signer correctly.
```
Comment (przydatek) [inline topics/icrc_29_window_post_message_transport.md:19]: also here, "user's machine"?

File: topics/icrc_29_window_post_message_transport.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+## Establishing the Communication Channel
```
Comment (przydatek) [inline topics/icrc_29_window_post_message_transport.md:21]: I'd say "Establishing a Communication Channel" (and start the paragraph below with "A communication channel"), but maybe ask a native speaker :-)

File: topics/icrc_29_window_post_message_transport.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+* The relying party can authenticate the signer by the origin of the window that it opens.
+* The singer can authenticate the relying party by the `origin` property of the `message` event that it received.
+
+## Sending Messages
+
+Messages are sent by calling `window.postMessage` on the signer window, or the `window.opener` respectively.
+When sending messages, the `targetOrigin` parameter must be set to the origin of the signer or relying party window.
+
+The relying party may close the signer window in between interactions. After sending a message, the relying party should
+wait for the signer to send a response before closing the window.
```
Comment (przydatek) [inline topics/icrc_29_window_post_message_transport.md:46]: What does it mean "the relying party should wait for the signer"?   What happens if it doesn't wait?

File: topics/icrc_29_window_post_message_transport.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+## Error Handling
+
+### Unexpectedly Closed Window
+
+If the signer window is closed unexpectedly, the relying party should handle this as if it had received a `NOT_GRANTED` error.
+Closing the window does _not_ end the session, and the relying party can open a new window and continue the session.
```
Comment (przydatek) [inline topics/icrc_29_window_post_message_transport.md:53]: this sentence talks about ending or continuing the session, but there is no mention in this document about opening a session.

**Review by przydatek — COMMENTED** [review]
Please see some minor suggestions and questions.

File: topics/icrc_29_window_post_message_transport.md (hunk 4 above)
Comment (frederikrothenberger) [reply topics/icrc_29_window_post_message_transport.md]: The result is undefined. 😉 I left a note about that.

File: topics/icrc_29_window_post_message_transport.md (hunk 5 above)
Comment (frederikrothenberger) [reply topics/icrc_29_window_post_message_transport.md]: Added a reference to the ICRC-25 session.

**Review by przydatek — APPROVED** [review]
LGTM, thanks!

**Pull request merged**